src/components/user/AccountTabs.vue

```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import BaseButton from "@/components/global/BaseButton.vue";
+
+const props = withDefaults(
+  defineProps<{
+    activeTab: string;
```

Comment:
When we know it's a strict list of known values, we should definitely provide hints on that and restrict our props as well.
```suggestion
    activeTab: 'personalDetails' | 'reservations';
```